asset-transfer-basic/application-gateway-java/src/main/java/App.java

```diff
@@ -34,22 +34,22 @@
 import java.util.concurrent.TimeUnit;
 
 public final class App {
-	private static final String mspID = "Org1MSP";
-	private static final String channelName = "mychannel";
-	private static final String chaincodeName = "basic";
+	private static final String MSP_ID = System.getenv().getOrDefault("MSP_ID", "Org1MSP");
+	private static final String CHANNEL_NAME = System.getenv().getOrDefault("CHANNEL_NAME", "mychannel");
+	private static final String CHAINCODE_NAME = System.getenv().getOrDefault("CHAINCODE_NAME", "basic");
 
 	// Path to crypto materials.
-	private static final Path cryptoPath = Paths.get("..", "..", "test-network", "organizations", "peerOrganizations", "org1.example.com");
+	private static final Path CRYPTO_PATH = Paths.get("../../test-network/organizations/peerOrganizations/org1.example.com");
```

Comment:
My years of working on systems running on windows and UNIX makes me twitch when I see I hard coded path separator :-)

Though perhaps it's not really a problem anymore?

Reply:
No problem, provided path sep is '/' not '\'.   _idiomatic Java_ is to use UPCASE for constants and '/' for path separators.